### DISCUSSION TITLE
I need help for human tracking ~

### DISCUSSION THREAD

**Discussion by songhat** [ultralytics/ultralytics#22240] [Q&A] [answered]
My task is to track people in a video, but because the video shakes violently, tracking is likely to fail. I understand that the tracking algorithm botsort supports re-identification, but the default feature recognition capability of the detector is not good. Does anyone have any recommended re-identification models?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @songhat, thank you for your interest in Ultralytics 🚀! This is an automated assistant response — an Ultralytics engineer will also review your discussion and assist soon.

For new users, we recommend visiting the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples and where many common questions are already answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom tracking ❓ Question, please share more details so we can help effectively:
-Your YOLO model and version (e.g., YOLO11 family), command(s) used, and tracker (e.g., BoT-SORT) settings
-Any custom `tracker` configuration values (conf, iou, reid settings), FPS, and frame size
-A short sample video or snippet showing the camera shake and failure cases
-Logs, screenshots, or output snippets illustrating ID switches or lost tracks
-Confirmation that you’ve reviewed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

Join the Ultralytics community where it suits you best:
-For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
-Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/)
-Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community

## Upgrade

Please upgrade to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to ensure you’re on the newest release:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can YOLO classification models as ReID model like `yolo11n-cls.pt`.

**Reply by songhat** [reply]
> You can YOLO classification models as ReID model like `yolo11n-cls.pt`.

Thank you for your reply. Can we get good performance just by using yolo11n-cls.pt without any fine-tuning?

**Reply by Y-T-G** [reply]
You can try. It's not going to be as good as dedicated ReID model.

**Reply by songhat** [reply]
Does integrating a dedicated ReID model require significant changes to the source code?

**Reply by glenn-jocher** [reply, marked as answer]
Not necessarily—no core changes: BoT-SORT already supports ReID via a tracker YAML; enable it and point `model` to your ReID weights (YOLO11 classification `.pt` or its TensorRT `.engine` are supported out of the box). For non‑YOLO ReID, a small wrapper replacing the `ReID` encoder to return per‑detection embeddings is sufficient; tracker logic stays unchanged. See the tracking guide and YAML parameters, and the ReID/BOTSORT reference for the encoder entry points.

yaml
# custom_botsort.yaml
tracker_type: botsort
with_reid: True
model: path/to/person_reid.engine  # or .pt
gmc_method: ecc  # try 'sparseOptFlow' for heavy camera shake

bash
yolo track model=yolo11n.pt source=video.mp4 tracker=custom_botsort.yaml

Docs: [Multi‑Object Tracking guide](https://docs.ultralytics.com/modes/track/), [botsort.yaml](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/trackers/botsort.yaml), [ReID/BOTSORT reference](https://docs.ultralytics.com/reference/trackers/bot_sort/).
